R/method-ops.R

```diff
@@ -16,7 +16,18 @@ on_load_define_ops <- function() {
 
 #' @export
 Ops.S7_object <- function(e1, e2) {
-  base_ops[[.Generic]](e1, e2)
+  cnd <- tryCatch(
+    return(base_ops[[.Generic]](e1, e2)),
+    methodNotFound = function(cnd) cnd
+  )
```

Comment:
I tried to think of any other way to do this to avoid the non-negligible performance hit of `tryCatch()`, but I can't really think of one:
- We don't have an easy to do `findMethod(.Generic, e1 = e1, e2 = e2)` so we can't try looking it up without calling it
- I don't think we can easily use the more performant `withCallingHandlers()` without more effort, since we don't always rethrow the error

Reply:
Given the general cost of dispatch I'm not sure this is worth worrying too much about?

``` r
library(S7)

foo <- new_class("foo")
method(`+`, list(foo, foo)) <- function(e1, e2) 100
f <- foo()

bench::mark(
  f + f,
  tryCatch(f + f)
)
#> # A tibble: 2 × 6
#>   expression           min   median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr>      <bch:tm> <bch:tm>     <dbl> <bch:byt>    <dbl>
#> 1 f + f             5.21µs   5.78µs   167089.    39.9KB     50.1
#> 2 tryCatch(f + f)   6.76µs   7.42µs   132355.    12.4KB     53.0
```

<sup>Created on 2023-11-29 with [reprex v2.0.2.9000](https://reprex.tidyverse.org)</sup>